SwiftyStoreKit/InAppReceipt.swift

```diff
@@ -253,65 +248,69 @@ internal class InAppReceipt {
             }
             task.resume()
     }
-
+  
     /**
-     *  Verify the purchase of a product in a receipt
+     *  Verify the purchase of a Consumable, NonConsumable, FreeSubscription or NonRenewingSubscription product in a receipt
      *  - Parameter productId: the product id of the purchase to verify
      *  - Parameter inReceipt: the receipt to test in
-     *  - Parameter validUntil: the expires date of the subscription must be valid until this date. If nil, no verification
      */
     class func verifyPurchase(
-      productId productId: String,
-      inReceipt receipt: ReceiptInfo,
-      purchaseType: SwiftyStoreKit.PurchaseType
+        productId productId: String,
+        inReceipt receipt: ReceiptInfo
     ) -> SwiftyStoreKit.VerifyPurchaseResult {
       
-        // Get all receipts
-        guard let allReceipts = receipt["receipt"]?["in_app"] as? [ReceiptInfo] else {
-            return .NotPurchased
-        }
+        // Get receipts info for the product
+        let receiptsInfo = self.getReceiptInfo(forProductId: productId, inReceipt: receipt)
       
-        // Filter receipts with matching product id
-        let receiptsMatchingProductId = allReceipts
-            .filter { (receipt) -> Bool in
-                let product_id = receipt["product_id"] as? String
-                return product_id == productId
-        }
+        // Verify that at least one receipt has the right product id
+        return receiptsInfo?.count >= 1 ? .Purchased : .NotPurchased
+    }
+  
+    /**
+     *  Verify the purchase of a AutomaticallyRenewableSubscription product in a receipt
+     *  - Parameter productId: the product id of the purchase to verify
+     *  - Parameter validUntil: the expires date of the subscription must be valid until this date. If nil, no verification
+     *  - Parameter inReceipt: the receipt to test in
+     */
+    class func verifyAutomaticallyRenewableSubscription(
+        productId productId: String,
+       validUntil date: NSDate = NSDate(),
```

Comment:
I believe it would be best if validUntil is the last argument as it's the only parameter with a default value. This way clients can omit it and `NSDate()` is used as default